pkg/plugins/steady/steady.go

```diff
@@ -0,0 +1,68 @@
+package steady
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/util/sets"
+	clusterapiv1 "open-cluster-management.io/api/cluster/v1"
+	clusterapiv1alpha1 "open-cluster-management.io/api/cluster/v1alpha1"
+	"open-cluster-management.io/placement/pkg/plugins"
+)
+
+const placementLabel = "cluster.open-cluster-management.io/placement"
+
+var _ plugins.Prioritizer = &Steady{}
+
+type Steady struct {
+	meta   metav1.ObjectMeta
+	handle plugins.Handle
+}
+
+func New(handle plugins.Handle) *Steady {
+	return &Steady{handle: handle}
+}
+
+func (s *Steady) Name() string {
+	return "steady"
+}
+
+func (s *Steady) Set(pl *clusterapiv1alpha1.Placement) {
+	s.meta = pl.ObjectMeta
+}
+
+func (s *Steady) Score(ctx context.Context, clusters []*clusterapiv1.ManagedCluster) (map[string]float64, error) {
+	// query placementdecisions with label selector
+	scores := map[string]float64{}
+	requirement, err := labels.NewRequirement(placementLabel, selection.Equals, []string{s.meta.Name})
+
+	if err != nil {
+		return nil, err
+	}
+
+	labelSelector := labels.NewSelector().Add(*requirement)
+	decisions, err := s.handle.ListDecisions(s.meta.Namespace, labelSelector)
+
+	if err != nil {
+		return nil, err
+	}
+
+	existingDecisions := sets.String{}
+	for _, decision := range decisions {
+		for _, d := range decision.Status.Decisions {
+			existingDecisions.Insert(d.ClusterName)
+		}
+	}
+
+	for _, cluster := range clusters {
+		if existingDecisions.Has(cluster.Name) {
+			scores[cluster.Name] = 1.0
```

Comment:
I think that this relative weighting means that balance will never outweigh steady.  I claim that enough imbalance should outweigh steadiness.  And that amount may vary by Placement, though such configuration could be a later objective.